### PR TITLE
 sysroot: Add ostree_sysroot_write_deployments_with_options()

### PR DESCRIPTION
**Pull request by cgwalters**
Depends: https://github.com/ostreedev/ostree/pull/744

More sophisticated users of libostree like rpm-ostree need control over things
like the system repository. Previously we introduced a "no cleanup" flag to
`ostree_sysroot_simple_write_deployment()`, but that's a high level API that
does filtering on its own.

Since rpm-ostree needs more control, let's expose the bare essentials of the
"sysroot commit" operation with an extensible options structure, where one of
the options is whether or not to do post-transaction repository operations.

### CODE REVIEW

**Comment by cgwalters** [comment]
🛫 Rebased 🛬

**Comment by cgwalters** [comment]
🛫 Rebased 🛬

File: src/libostree/ostree-sysroot-deploy.c [hunk 1]
```diff
@@ -1686,25 +1686,41 @@ is_ro_mount (const char *path)
  * @cancellable: Cancellable
  * @error: Error
  *
- * Assuming @new_deployments have already been deployed in place on
- * disk, atomically update bootloader configuration.
+ * Older verison of ostree_sysroot_write_deployments() - will perform
```
Comment (jlebon) [inline src/libostree/ostree-sysroot-deploy.c]: s/verison/version/

Comment (jlebon) [inline src/libostree/ostree-sysroot-deploy.c]: s/`ostree_sysroot_write_deployments`/`ostree_sysroot_write_deployments_with_options`/ ?

File: src/libostree/ostree-sysroot-deploy.c [hunk 2]
```diff
@@ -1686,25 +1686,41 @@ is_ro_mount (const char *path)
...
+ * ostree_sysroot_write_deployments_with_options:
+ * @self: Sysroot
+ * @new_deployments: (element-type OstreeDeployment): List of new deployments
+ * @opts: Options
+ * @cancellable: Cancellable
+ * @error: Error
+ *
+ * Assuming @new_deployments have already been deployed in place on disk via
+ * ostree_sysroot_deploy_tree(), atomically update bootloader configuration. By
+ * default, no-post-transaction cleanup will be performed. You should invoke
```
Comment (jlebon) [inline src/libostree/ostree-sysroot-deploy.c]: `s/no-/no /`

Comment (cgwalters) [inline src/libostree/ostree-sysroot-deploy.c]: Adding a comment here that I've addressed your comments on the code comments ⬇️

**Comment by jlebon** [comment]
@rh-atomic-bot r+ ae59f35

**Comment by cgwalters** [comment]
@rh-atomic-bot retry

**Pull request closed without merging**